src/utils/utils.js

```diff
@@ -12,10 +13,46 @@ export function handleImageClick(thisObj) {
   const title = thisObj.name;
   const link = thisObj.link;
 
-  cardPopup.open({ title, link });
+  previewPopup.open({ title, link });
 }
-
+/*
 export function createCard(item) {
   const card = new Card(item, "#card-template", handleImageClick);
   return card.generateCard();
 }
+*/
+export function createCard(item) {
+  const card = new Card(item, "#card-template", handleImageClick, {
+    handleDelete: (cardElement, id) => {
+      deletePopup.open();
+      deletePopup.catchSelectedCard(cardElement, id);
+    },
+    handleLike: (id) => {
+      api.addLike(id).then((r) => {
+        card.handleLikeButton();
+      });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere**